wrpcorrectify/metrics.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: 2022 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package wrpcorrectify
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/xmidt-org/touchstone"
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+// CounterMetric provides a counter metric that can be used to track the number
+// of times a specific error occurs.
+type CounterMetric struct {
```

Comment:
This code might benefit from inverting the metric function.  You can avoid any dependency on `prometheus`.  Instead, expose methods that return whatever state you want to track, then require application code to use the `*Func` versions of metrics.  For example: https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#NewCounterFunc.

Reply:
I like your suggestion better.